fixes #4111 - Add image user data flag to API

shlomizadok opened the pull request:
fixes #4111

ohadlevy commented:
[test]

isratrade commented:
@shlomizadok, looks good. Can you squash the two commit into one and have only one line break at end of file.

domcleal commented on app/views/api/v2/images/main.json.rabl:
```diff
@@ -5,5 +5,10 @@ extends "api/v2/images/base"
 attributes :operatingsystem_id, :operatingsystem_name, :compute_resource_id, :compute_resource_name,
            :architecture_id, :architecture_name, :uuid, :username, :created_at, :updated_at
 
+attribute :user_data, :if => lambda { |img| (img.compute_resource.kind_of? Foreman::Model::EC2) ||
```
This isn't very flexible, it should use fragments per compute resource like app/views/api/v2/compute_resources/ so plugins can extend it.

shlomizadok replied:
@domcleal -- good point. refactoring.

shlomizadok commented:
Changed so is can be more flexible. I have added an if that checks whether the provider is EC2, Rackspace, OpenStack so we won't have to add a (blank) file per each provider. cc @domcleal

ohadlevy commented on app/views/api/v2/images/main.json.rabl:
```diff
@@ -5,5 +5,14 @@ extends "api/v2/images/base"
 
+
+
+if @image && ((@image.compute_resource.kind_of? Foreman::Model::EC2) || (@image.compute_resource.kind_of? Foreman::Model::Rackspace) || (@image.compute_resource.kind_of? Foreman::Model::Openstack))
```
this is not portable, e.g. a plugin cant extend / use it.

another option could be to add a method on the compute resource, similar to capability ?

isratrade commented:
@shlomizadok, looks good.  Squash the 3 commits into 1 and then it looks ready for merge.

dLobatog commented on test/unit/compute_resource_test.rb:
```diff
@@ -102,4 +102,9 @@ def setup
   end
 
+  test "user_data_supported?" do
+    assert_equal false, compute_resources(:one).user_data_supported?
```
Please use `refute` and `assert` instead of `assert_equal boolean`

shlomizadok replied:
:+1:

dLobatog commented:
Reviewed and tested, to me it's nearly ready to merge, @shlomizadok, please remove doc/apipie_examples.json from the commit.

shlomizadok committed:
fixes #4111 - Add image user data flag to API

isratrade commented:
merged as d21103bcf13b5981601be88330ce73dbe4a1ed77. Thanks @shlomizadok !
